dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/provisioning/ServiceFingerprint.java

```diff
@@ -226,13 +200,16 @@ public static void putInstallDir(String name, File installDir) {
             name = "core";
         }
 
-        internalPut(new File(installDir, "lib/versions/service.fingerprint"));
-        try {
-            props.setProperty(name, installDir.getCanonicalPath());
-        } catch (IOException ioe) {
-            // ignore this. If it happens for core we will do an auto clean next time, otherwise we don't really care.
+        if (!props.containsKey(name)) {
+            internalPut(new File(installDir, "lib/versions/service.fingerprint"));
```

Comment:
I don't understand the check here.  So if the install dir name (e.g. "core") is already in the prop we loaded from disk we don't need to recheck its service.fingerprint hash to make sure it is not different from when we loaded the props?